### ISSUE TITLE
[Policy Router] Integrate multiple ventillators and add service discovery

### ISSUE THREAD

**Issue by gaurav-teliax** [tfxc/issues#23]
Right now the Ventilator is a single point of failure in TFXC project. To make it HA we will be adding multiple ventilator instances. 
We will be adding service discovery with Consul as well.

**Comment by gaurav-teliax** [comment]
Chris => David

> To enable better availability and resilience to problems for the TFXC it
> will be necessary to have multiple Ventilators concurrently. For the PR,
> this means it will need to support connecting to multiple Ventilators.
> 
> The only requirement for PR is that it sends responses to queries out
> through the same Ventilator it received them (but not necessarily the
> same socket). This means it won't be possible to make multiple
> connections with the same ZMQ socket since it's not possible to control
> the destination Ventilator in that case.
> 
> The plan is to use have Ventilator register using Consul which can do
> health checks and return a list of healthy Ventilators when queried.
> However, the PR probably doesn't really care about which Ventilators are
> healthy and can connect to all of them. It's unlikely a request will
> come through from a Ventilator that is down.
> 
> While dynamically getting the list of Ventilators from Consul would be
> the ideal case, in the short term PR could connect to a preconfigured
> list of all Ventilators. PR doesn't necessarily even need to support
> using multiple Ventilators in that case as a completely separate PR
> could be run for each one (this isn't ideal but it would work for having
> HA with that part of the TFXC quickly).
> 
> Please let me know if there are any problems/questions.

**Comment by Bruno-me** [comment]
>This means it won't be possible to make multiple connections with the same ZMQ socket since it's not possible to control the destination Ventilator in that case.

This is not correct. The zmq '[ROUTER](http://api.zeromq.org/2-1:zmq-socket#toc7)' socket type allows specifying a destination peer identity when sending a messsage:

>When sending messages a ZMQ_ROUTER socket shall remove the first part of the message and use it to determine the identity of the peer the message shall be routed to.

**Comment by vulpyne** [comment]
> This is not correct. The zmq '[ROUTER](http://api.zeromq.org/2-1:zmq-socket#toc7)' socket type allows specifying a destination peer identity when sending a messsage:

Right, we already use ROUTER sockets for some stuff. However, some of the sockets are PUSH/PULL since they only need to load balance.

Also, switching to using ROUTER sockets for everything would add a fair amount of complexity and it would be necessary to have some sort of discovery on both sides to figure out what peers to route to. Not really worth it.

Another problem with the multiple connections with same ZMQ socket approach is that while you can add endpoints you currently don't seem to be able to remove them (at least on the Lua side). It's easier to just use separate sockets and poll. 

In the PR case it can treat each Ventilator it's connected to like a completely separate process so it's definitely simpler that way.
